applications/FluidDynamicsApplication/python_scripts/fluid_solver.py

```diff
@@ -259,3 +273,162 @@ def _SetNodalProperties(self):
         KratosMultiphysics.VariableUtils().SetVariable(KratosMultiphysics.VISCOSITY, kin_viscosity, self.main_model_part.Nodes)
+
+    # TODO: I THINK THIS SHOULD BE MOVED TO THE BASE PYTHON SOLVER
+    def is_restarted(self):
+        # this function avoids the long call to ProcessInfo and is also safer
+        # in case the detection of a restart is changed later
+        return self.main_model_part.ProcessInfo[KratosMultiphysics.IS_RESTARTED]
+
+    def GetEstimateDtUtility(self):
+        if not hasattr(self, '_estimate_dt_utility'):
+            self._estimate_dt_utility = self._CreateEstimateDtUtility()
+        return self._estimate_dt_utility
+
+    def _GetScheme(self):
+        if not hasattr(self, '_scheme'):
+            self._scheme = self._CreateScheme()
+        return self._scheme
+
+    def GetConvergenceCriterion(self):
+        if not hasattr(self, '_convergence_criterion'):
+            self._convergence_criterion = self._CreateConvergenceCriterion()
+        return self._convergence_criterion
+
+    def _GetLinearSolver(self):
+        if not hasattr(self, '_linear_solver'):
+            self._linear_solver = self._CreateLinearSolver()
+        return self._linear_solver
+
+    def _GetBuilderAndSolver(self):
+        if not hasattr(self, '_builder_and_solver'):
+            self._builder_and_solver = self._CreateBuilderAndSolver()
+        return self._builder_and_solver
+
+    def GetSolutionStrategy(self):
+        if not hasattr(self, '_solution_strategy'):
+            self._solution_strategy = self._CreateSolutionStrategy()
+        return self._solution_strategy
+
+    def _CreateEstimateDtUtility(self):
+        domain_size = self.GetComputingModelPart().ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        if domain_size == 2:
+            estimate_dt_utility = KratosCFD.EstimateDtUtility2D(
+                self.GetComputingModelPart(),
+                self.settings["time_stepping"])
+        else:
+            estimate_dt_utility = KratosCFD.EstimateDtUtility3D(
+                self.GetComputingModelPart(),
+                self.settings["time_stepping"])
+
+        return estimate_dt_utility
+
+    def _CreateScheme(self):
+        domain_size = self.GetComputingModelPart().ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        # Cases in which the element manages the time integration
+        if self.element_integrates_in_time:
+            # "Fake" scheme for those cases in where the element manages the time integration
+            # It is required to perform the nodal update once the current time step is solved
+            scheme = KratosMultiphysics.ResidualBasedIncrementalUpdateStaticSchemeSlip(
+                domain_size,
+                domain_size + 1)
+            # In case the BDF2 scheme is used inside the element, the BDF time discretization utility is required to update the BDF coefficients
+            if (self.settings["time_scheme"].GetString() == "bdf2"):
```

Comment:
This class ("FluidSolver") has no parameters O,O

This is quite a bad design IMO :sweat_smile: 

If those settings are common to all FluidSolvers the settings should be here
If not, then those functions should not be here but instead in some aux-files

Reply:
The point is that the design of the fluid solvers is based on the formulation / strategy, as was there before the json, stages, processes, etc. This makes really hard to define common parameters.

I'm aware of this, however this would imply to completely refactor the settings in all solvers. Aside of being a huge amount of work, which spreads to all the GUIs.

If required, I'd open an issue to discuss about as this would affect Chimera, CompressibleFlow, ConvectionDiffusion, FSI and possibly more applications.

Reply:
(added to the Fluid Dynamics Application project)